OathLab_Twitter/app/src/main/java/net/serkanbal/oathlab_twitter/MainActivity.java

```diff
@@ -0,0 +1,178 @@
+package net.serkanbal.oathlab_twitter;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+import com.google.gson.Gson;
+
+import org.apache.commons.codec.binary.Base64;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.FormBody;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+public class MainActivity extends AppCompatActivity {
+    public static final String CONSUMER_KEY = "gKOs6rrJhWoH73ZHTjyy33QFH";
+    public static final String CONSUMER_SECRET = "0LF56WARaf02v1KilCfBnpCZAsG2HBAnPAdLQu377FO7BA7vVn";
+    public String mScreenName = "MargotRobbie";
+    List<TwitterObject> mList = new ArrayList<>();
+    EditText mScreenNameEntry;
+    Button mSearch;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        getBearerToken(encodeKeys(CONSUMER_KEY, CONSUMER_SECRET));
+
+        mSearch = (Button) findViewById(R.id.search_button);
+        mScreenNameEntry = (EditText) findViewById(R.id.search_query);
+
+        mSearch.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                mScreenName = mScreenNameEntry.getText().toString();
+                getBearerToken(encodeKeys(CONSUMER_KEY, CONSUMER_SECRET));
```

Comment:
You could just store it in a member variable. Shared prefs would also be good, because then if the user navigates to another activity and comes back, you wouldn't necessarily have to get a new bearer token.

Unfortunately, the only way to find out if a bearer token is valid is to use it in a request. It's up to Twitter to deem the token valid or invalid, so there's no way to test the validity without making a call to Twitter. The [api docs](https://dev.twitter.com/oauth/application-only) describe the response they'll give if the bearer token is expired under the "API request contains invalid bearer token" section. So, you'd have to check if that response was received